databese design

motemotenten opened the pull request:
# WHAT
データベースの設計、リレーションの確認

# WHY
コードを書くまでにデータベースを設計する必要があるから

# HOW
マークダウン記法を使って、多対多has_many throughを考えてリレーションを定義、データベースの設計

motemotenten committed:
databese design

snowhork commented on README.md:
```diff
@@ -0,0 +1,76 @@
+
+* Configuration
+
+* Database creation
```
この辺りまで消しちゃっていいんじゃないかな？

snowhork commented on README.md:
```diff
@@ -0,0 +1,76 @@
+|group_id | *integer*
+
+#### ※messagesテーブルのbodyとimageカラム以外null制約をつける
+#### ※groupsテーブルのnameカラムとusersテーブルのnameカラムにindexを貼る
```
オプションは各テーブルに一列増やして、書いておきましょう。

snowhork commented on README.md:
```diff
@@ -0,0 +1,76 @@
+|id |*integer*
+|name|*string*
+
+## Mediumテーブル
```
中間テーブルは、groups_users テーブルにしておきましょうか

snowhork commented on README.md:
```diff
@@ -0,0 +1,76 @@
+    has_many :users, through: :mediums
+#### medium.rb
+    belongs_to :user
+    belongs_to :group
```
アソシエーションも、それぞれのテーブルのカラム一覧の下に書いておこう！

snowhork commented on README.md:
```diff
@@ -0,0 +1,76 @@
+
+
+Please feel free to use a different markup language if you do not plan to run
+<tt>rake doc:app</tt>.
```
この辺りも削除で！

motemotenten committed:
modify database design in README.md

・change midium to groups_users
・remove unnecessary code
・rewrite association and options

Maverick1211 commented:
・medium→groups_users
・余分な記述の消去
・アソシエーションとオプションを書く場所

以上四点修正しました！

snowhork commented on README.md:
```diff
@@ -1,76 +1,53 @@
-* Database creation
 ## Usersテーブル
-|column      |  type|
+|column      |  type|option
```
多分、ここのoptionに　| が付いてないせいか、３行目にもう一列足りないせいか、markdownが崩れてます。

githubでは、各ファイルの右にあるViewというボタンをクリックするとmarkdownでどう当たってるか確認できます！

snowhork commented on README.md:
```diff
@@ -1,76 +1,53 @@
+|confirm password |*string*|null false
+|group_id |*integer*|null false
+
+#### user.rb
```
****.rbはなくていいかな！(アソシエーションもテーブル設計の一部と考えていいと思います

motemotenten committed:
modify database design in README.md 2nd

・group,rb→group association
・|option→|option|

motemotenten committed:
modify database design in README.md 3rd

motemotenten committed:
modify database design in README.md 4th

Maverick1211 commented:
・|を追加して見た目を修正しました！

snowhork commented on README.md:
```diff
@@ -0,0 +1,53 @@
+
+#### user association
+    has_many :messages
+    has_many :groups, through: :mediums
```
中間テーブル名前変えたよね！

snowhork replied:
has_many :中間テーブル
も必要かな？

motemotenten committed:
change model name

snowhork commented on README.md:
```diff
@@ -0,0 +1,55 @@
+|confirm password |*string*|null false    |
+|group_id |*integer*|null false           |
+
+#### user association
```
細かいけど、associationだけで良いと思う！

snowhork commented on README.md:
```diff
@@ -0,0 +1,55 @@
+|column      |  type|option               |
+|:-----------|-----------|-----------:|
+|id               |*integer*|null false   |
+|name         | *string*|null false ※add_index :users, :name|
```
`※add_index :users, :name`
これはどういうことだろう？
nameカラムにindexを追加するなら、add_indexだけでいいと思う！

snowhork commented on README.md:
```diff
@@ -0,0 +1,55 @@
+|column      |  type|option           |
+|:-----------|-----------|-----------:|
+|id |*integer*|null false             |
+|name|*string*|null false ※add_index :users, :name|
```
※add_index :users, :name
これも同様に変だね！

motemotenten committed:
modify few points

snowhork commented on README.md:
```diff
@@ -2,13 +2,13 @@
 |:-----------|-----------|-----------:|
 |id               |*integer*|null false   |
-|name         | *string*|null false ※add_index :users, :name|
+|name         | *string*|null false ※add_index :name|
```
:name　も要らないかな？

motemotenten committed:
modify few points
